### PR TITLE
Step10 core view topics

### PR DESCRIPTION
**Pull request by mikef80**
implemented topic selection

### CODE REVIEW

**Comment by mikef80** [comment]
just realised it's not working for non-existent topics.  Looking at fix now

File: src/api/api.js [hunk 1]
```diff
@@ -4,13 +4,15 @@ const ncNewsApi = axios.create({
...
-    data.articles.sort(
-      (a, b) => new Date(b.created_at) - new Date(a.created_at)
-    );
-    return data;
-  });
+export const getAllArticles = (topic) => {
+  return ncNewsApi
+    .get("/articles", { params: { topic: topic } })
+    .then(({ data }) => {
+      data.articles.sort(
```
Comment (ghost) [inline src/api/api.js]: This sorting should have been implemented in your back end.

File: src/components/TopicListItem.jsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+          <img
+            src={articles[3].article_img_url}
+            alt=''
+            className='rounded-br-xl w-[100px] aspect-square object-cover'
+          />
+          <div className='absolute top-1/2 -translate-y-1/2  text-3xl bg-white rounded-3xl p-2 border-2 border-black'>
+            {slug}
+          </div>
+      </li>
+        </Link>
```
Comment (ghost) [inline src/components/TopicListItem.jsx:46]: Took me a second to realise why you were doing the api call in here, nicely done.

**Review by ghost — APPROVED** [review]
Nice job, url is updated and sorting appears to work well.

File: src/api/api.js (hunk 1 above)
Comment (mikef80) [reply src/api/api.js]: It is, so I don't know why I put that there.  Have tested just to make sure, so will delete it and merge

**Pull request merged**